tools/codegen.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+
+sqlite=$1
+postgresql=$2
+
+set -e
+
+echo "Codegen: SQLITE: $sqlite, POSTGRES: $postgresql"
+
+lib_dir=gen/lib_of_uri
+mkdir -p $lib_dir
+
+lib_ml=$lib_dir/trakeva_of_uri.ml
+lib_mli=$lib_dir/trakeva_of_uri.mli
+
+cat <<EOBLOB > $lib_ml
+
+open Pvem_lwt_unix
+open Pvem_lwt_unix.Deferred_result
+
+module Trakeva_missing : Trakeva.KEY_VALUE_STORE = struct
+   type t
+   let load s =
+     Printf.ksprintf failwith "No trakeva backend can understand: %S" s
+   let get ?collection t ~key = assert false
+   let get_all _ = assert false
+   let act _ = assert false
+   let iterator _ = assert false
+   let close _ = assert false
+end
+EOBLOB
+
+if [ "$sqlite" = "true" ]; then
+    echo "module Sqlite = Trakeva_sqlite" >> $lib_ml
+else
+    echo "module Sqlite = Trakeva_missing" >> $lib_ml
+fi
+if [ "$postgresql" = "true" ]; then
+    echo "module Postgresql = Trakeva_postgresql" >> $lib_ml
+else
+    echo "module Postgresql = Trakeva_missing" >> $lib_ml
+fi
+
+cat <<EOBLOB >> $lib_ml
+type t = {
+  implementation: (module Trakeva.KEY_VALUE_STORE with type t = t);
+}
+let load s =
+  let uri = Uri.of_string s in
+  let backend =
+     match Uri.scheme uri with
+     | Some "postgresql" -> (module Postgresql : Trakeva.KEY_VALUE_STORE)
+     | Some "sqlite" | None -> (module Sqlite : Trakeva.KEY_VALUE_STORE)
```

Comment:
Why default to `Sqlite` if can't figure out uri? Shouldn't `trakeva_of_uri` then fail instead?